modules/videoio/test/test_ffmpeg.cpp

```diff
@@ -468,7 +468,7 @@ static void ffmpeg_check_read_raw(VideoCapture& cap)
     cap >> data;
     EXPECT_EQ(CV_8UC1, data.type()) << "CV_8UC1 != " << typeToString(data.type());
     EXPECT_TRUE(data.rows == 1 || data.cols == 1) << data.size;
-    EXPECT_EQ((size_t)29729, data.total());
+    EXPECT_EQ((size_t)29774, data.total());
```

Comment:
Changed behavior.

This patch probably emit broken stream.

Reply:
highgui/video/big_buck_bunny.mp4 cannot be decoded inside cudacodec without the additional information from parsing the mpeg4 header.
If you have access to the lastet nvidia video codec sdk samples they have included this information for MPEG4 decoding inside Samples\Utils\FFmpegDemuxer.h, extract below
```
if (bMp4MPEG4 && (frameCount == 0)) {

    int extraDataSize = fmtc->streams[iVideoStream]->codecpar->extradata_size;

    if (extraDataSize > 0) {

        // extradata contains start codes 00 00 01. Subtract its size
        pDataWithHeader = (uint8_t *)av_malloc(extraDataSize + pkt.size - 3*sizeof(uint8_t));

        if (!pDataWithHeader) {
            LOG(ERROR) << "FFmpeg error: " << __FILE__ << " " << __LINE__;
            return false;
        }

        memcpy(pDataWithHeader, fmtc->streams[iVideoStream]->codecpar->extradata, extraDataSize);
        memcpy(pDataWithHeader+extraDataSize, pkt.data+3, pkt.size - 3*sizeof(uint8_t));

        *ppVideo = pDataWithHeader;
        *pnVideoBytes = extraDataSize + pkt.size - 3*sizeof(uint8_t);
    }

} else {
    *ppVideo = pkt.data;
    *pnVideoBytes = pkt.size;
}
```
althouth the decoder works without removing the start code in this case.